packages/api/src/routes/medical/document.ts

```diff
@@ -167,8 +170,11 @@ async function getDownloadUrl(req: Request): Promise<string> {
     throw new ForbiddenError(message); // This should be 404
   }
 
-  const url = await downloadDocument({ fileName: fileNameString, conversionType });
-  return url;
+  if (fileNameString.startsWith("location=hl7/")) {
+    return await getIncomingHl7MessageDownloadUrl({ fileName: fileNameString });
+  }
+
+  return await getDocumentDownloadUrl({ fileName: fileNameString, conversionType });
```

Comment:
Let's avoid logic on the route if we can, please

Reply:
Also, `getDocumentDownloadUrl` has logic for `hl7` conversion type, so do we really need this additional check?